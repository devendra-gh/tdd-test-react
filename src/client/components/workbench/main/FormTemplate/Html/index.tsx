import React from 'react';

import { noop, isFunction } from 'lodash';

import Checkbox from '@tamm/ui-lib-v2-checkbox';
import CheckboxGroup from '@tamm/ui-lib-v2-checkbox-group';
import DatePicker from '@tamm/ui-lib-v2-date-picker';
import Input from '@tamm/ui-lib-v2-input';
import InputNumber from '@tamm/ui-lib-v2-input-number';
import InputPassword from '@tamm/ui-lib-v2-input-password';
import InputTelephone from '@tamm/ui-lib-v2-input-telephone';
import RadioGroup from '@tamm/ui-lib-v2-radio-group';
import RangeInput from '@tamm/ui-lib-v2-range-input';
import DlsButton from '@tamm/ui-lib-v2-button';
import DlsForm from '@tamm/ui-lib-v2-form';
import Select from '@tamm/ui-lib-v2-select';
import Slider from '@tamm/ui-lib-v2-slider';
import TextArea from '@tamm/ui-lib-v2-text-area';
import TimePicker from '@tamm/ui-lib-v2-time-picker';
import Grid from '@tamm/ui-lib-v2-grid';
import { IVariables } from '@tamm/app-composer';

import FileUpload from '../../FileUpload';
import { FieldErrorType } from '../Types';

const CheckboxInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <>
      <Checkbox
        {...field.props}
        onChange={handleInputChange(field)}
        checked={fieldValue}
        validateStatus={fieldError ? 'error' : undefined}
        disabled={disabled}
      />
    </>
  );
};

const CheckboxGroupInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <>
      <CheckboxGroup
        {...field.props}
        onChange={handleInputChange(field)}
        i18n={noop}
        disabled={disabled}
      />
    </>
  );
};

const DatePickerInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <DatePicker
      {...field.props}
      i18n={noop}
      onChange={handleInputChange(field)}
      validateStatus={fieldError ? 'error' : undefined}
      help={
        (fieldError || [])
          .map((error: FieldErrorType) => error.message)
          .join('. ') || field.props.help
      }
      disabled={disabled}
      value={fieldValue}
    />
  );
};

const TextInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <>
      <Input
        {...field.props}
        onChange={handleInputChange(field)}
        value={fieldValue}
        validateStatus={fieldError ? 'error' : undefined}
        disabled={disabled}
        help={
          (fieldError || [])
            .map((error: FieldErrorType) => error.message)
            .join('. ') || field.props.help
        }
      />
    </>
  );
};

const NumberInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <>
      <InputNumber
        {...field.props}
        onChange={handleInputChange(field)}
        value={fieldValue}
        validateStatus={fieldError ? 'error' : undefined}
        disabled={disabled}
        help={
          (fieldError || [])
            .map((error: FieldErrorType) => error.message)
            .join('. ') || field.props.help
        }
      />
    </>
  );
};

const PasswordInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <>
      <InputPassword
        {...field.props}
        onChange={handleInputChange(field)}
        value={fieldValue}
        validateStatus={fieldError ? 'error' : undefined}
        disabled={disabled}
        help={
          (fieldError || [])
            .map((error: FieldErrorType) => error.message)
            .join('. ') || field.props.help
        }
      />
    </>
  );
};

const TelephoneInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <>
      <InputTelephone
        {...field.props}
        onSelect={handleInputChange(field)}
        value={fieldValue}
        validateStatus={fieldError ? 'error' : undefined}
        disabled={disabled}
        help={
          (fieldError || [])
            .map((error: FieldErrorType) => error.message)
            .join('. ') || field.props.help
        }
      />
    </>
  );
};

const TimePickerInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <>
      <TimePicker
        {...field.props}
        value={fieldValue}
        onChange={handleInputChange(field)}
        validateStatus={fieldError ? 'error' : undefined}
        help={
          (fieldError || [])
            .map((error: FieldErrorType) => error.message)
            .join('. ') || field.props.help
        }
        disabled={disabled}
      />
    </>
  );
};

const RadioInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  const items = (field.props.items || []).map((item: IVariables) => {
    if (fieldValue) {
      if (item.label === fieldValue) return { ...item, checked: true };
      return { ...item, checked: false };
    }
    return item;
  });
  return (
    <RadioGroup
      {...field.props}
      items={items}
      onChange={handleInputChange(field)}
      value={fieldValue}
      validateStatus={fieldError ? 'error' : undefined}
      disabled={disabled}
      help={
        (fieldError || [])
          .map((error: FieldErrorType) => error.message)
          .join('. ') || field.props.help
      }
    />
  );
};

const InputRange = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <RangeInput
      {...field.props}
      onChange={handleInputChange(field)}
      value={fieldValue}
      validateStatus={fieldError ? 'error' : undefined}
      disabled={disabled}
      help={
        (fieldError || [])
          .map((error: FieldErrorType) => error.message)
          .join('. ') || field.props.help
      }
    />
  );
};

const SelectInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;

  return (
    <Select
      {...field.props}
      onChange={handleInputChange(field)}
      value={fieldValue}
      multi={field.multi || false}
      disabled={disabled}
      validateStatus={fieldError ? 'error' : undefined}
      help={
        (fieldError || [])
          .map((error: FieldErrorType) => error.message)
          .join('. ') || field.props.help
      }
    />
  );
};

const TextAreaInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <TextArea
      {...field.props}
      onChange={handleInputChange(field)}
      value={fieldValue}
      disabled={disabled}
      validateStatus={fieldError ? 'error' : undefined}
      help={
        (fieldError || [])
          .map((error: FieldErrorType) => error.message)
          .join('. ') || field.props.help
      }
    />
  );
};

const FileInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <FileUpload
      {...field.props}
      i18n={noop}
      disabled={disabled}
      validateStatus={fieldError ? 'error' : undefined}
      validationMessage={
        (fieldError || [])
          .map((error: FieldErrorType) => error.message)
          .join('. ') || field.props.validationMessage
      }
      files={fieldValue || []}
      onChange={handleInputChange(field)}
      onRemove={handleInputChange(field)}
    />
  );
};

const SliderInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <>
      <Slider
        {...field.props}
        onChange={handleInputChange(field)}
        value={fieldValue}
        validateStatus={fieldError ? 'error' : undefined}
        disabled={disabled}
        help={
          (fieldError || [])
            .map((error: FieldErrorType) => error.message)
            .join('. ') || field.props.help
        }
      />
    </>
  );
};

const Form = ({
  children,
  handleCancel,
  handleSubmit,
  handleBack,
  onSubmitLabel,
  onBackLabel,
  title,
  props,
  includeCancelButton,
  includeBackButton,
  btnSubmitArrow,
  btnBackArrow,
  formValid,
  defaultSubmitLabel,
  defaultBackLabel,
  defaultCancelLabel,
}: any) => {
  return (
    <div>
      <DlsForm
        backButton={
          includeBackButton
            ? {
                label: onBackLabel || defaultBackLabel,
                onClick: (e: React.FormEvent<HTMLFormElement>) => {
                  // eslint-disable-next-line no-unused-expressions
                  handleBack && isFunction(handleBack) && handleBack(e, props);
                },
                withArrow: btnBackArrow,
                alignIcon: btnBackArrow,
              }
            : undefined
        }
        submitButton={{
          label: onSubmitLabel || defaultSubmitLabel,
          onClick: (e: React.FormEvent<HTMLFormElement>) => {
            handleSubmit(e, props);
          },
          withArrow: btnSubmitArrow,
          alignIcon: btnSubmitArrow,
          disabled: !formValid,
        }}
        cancelLink={
          includeCancelButton
            ? {
                label: defaultCancelLabel,
                href: '',
                onClick: (e: React.FormEvent<HTMLFormElement>) => {
                  // eslint-disable-next-line no-unused-expressions
                  handleCancel &&
                    isFunction(handleCancel) &&
                    handleCancel(e, props);
                },
              }
            : undefined
        }
      >
        {children}
      </DlsForm>
    </div>
  );
};

const FieldGroup = ({ field, children, width }: any) => {
  return <div key={field.name}>{children}</div>;
};

const SectionForm = ({ title, children }: any) => {
  return null;
};

const Button = ({ field, handleInputChange }: any) => {
  return (
    <DlsButton {...field.props} aria-label="button-primary" size="small" />
  );
};

export {
  Button,
  CheckboxInput,
  CheckboxGroupInput,
  DatePickerInput,
  FieldGroup,
  FileInput,
  Form,
  Grid,
  InputRange,
  NumberInput,
  PasswordInput,
  RadioInput,
  SelectInput,
  SectionForm,
  TextInput,
  TelephoneInput,
  TextAreaInput,
  TimePickerInput,
  SliderInput,
};
