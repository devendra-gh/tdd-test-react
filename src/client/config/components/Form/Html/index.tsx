import React from 'react';
import Button from '@tamm/ui-lib-v2-button';
// import Checkbox from '@tamm/ui-lib-v2-checkbox';
// import DatePicker from '@tamm/ui-lib-v2-date-picker';
import Form from '@tamm/ui-lib-v2-form';
import Input from '@tamm/ui-lib-v2-input';
import Label from '@tamm/ui-lib-v2-label';
import Select from '@tamm/ui-lib-v2-select';
// import moment from 'moment';
import RadioGroup from '@tamm/ui-lib-v2-radio-group';
import Container from 'client/containers';
import { SelectItemType, FieldErrorType } from '../Types';

const FieldErrorMessage = ({
  fieldErrors,
  i18n,
}: {
  fieldErrors: FieldErrorType[];
  i18n: Function;
}): any => {
  return fieldErrors
    ? fieldErrors.map((error: FieldErrorType) => (
        <span style={{ color: '#ae1f23' }}>{i18n(error.message)}</span>
      ))
    : '';
};

const CheckboxInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
  i18n,
}: any) => {
  // const { disabled } = field;

  return (
    <>
      {/* {field.label && <Label>{field.label}</Label>} */}
      {Array.isArray(field.options) &&
        field.options.map((option: any) => (
          // <BsForm.Check
          //   type="checkbox"
          //   label={option.label}
          // onChange={handleInputChange(field)}
          // name={field.name}
          // value={option.value}
          // disabled={disabled}
          // checked={fieldValue && fieldValue[option.value]}
          //   isInvalid={!!fieldError}
          // />
          // <Checkbox
          //   label={i18n(field.label)}
          //   onChange={handleInputChange(field)}
          //   name={field.name}
          //   value={option.value}
          //   disabled={disabled}
          //   checked={fieldValue && fieldValue[option.value]}
          // />
          <div>checkbox</div>
        ))}
      <FieldErrorMessage fieldErrors={fieldError} i18n={i18n} />
    </>
  );
};

const RadioInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
  i18n,
  questionnaire,
}: any) => {
  const { disabled } = field;
  const label = () => {
    if (questionnaire && field.title) {
      return <div className="questionnaire__label">{i18n(field.title)}</div>;
    }
    return field.label && <Label>{i18n(field.label)}</Label>;
  };
  return (
    <div style={{ marginBottom: questionnaire && '3rem' }}>
      {label()}
      <RadioGroup i18n={i18n} align={field.align} uiType={field.uiType}>
        {field.options.map((option: any) => (
          <RadioGroup.Radio
            label={i18n(option.label)}
            onChange={handleInputChange(field)}
            name={option.label}
            value={option.value}
            disabled={disabled}
            checked={option.value === fieldValue}
            description={option.description && i18n(option.description)}
          />
        ))}
      </RadioGroup>
      <FieldErrorMessage fieldErrors={fieldError} i18n={i18n} />
    </div>
  );
};

const SelectInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
  i18n,
}: any) => {
  const { disabled } = field;
  const items = field.options.map((option: SelectItemType) => ({
    id: option.value,
    label: i18n(option.label),
  }));

  return (
    <>
      {/* <BsForm.Control
        as={field.type}
        name={field.name}
        value={fieldValue}
        onChange={handleInputChange(field)}
        disabled={disabled}
        isInvalid={!!fieldError}
      >
        {field.options &&
          field.options.map((option: SelectItemType) => {
            return (
              <option key={option.value} value={option.value}>
                {option.label}
              </option>
            );
          })}
        <FieldErrorMessage fieldErrors={fieldError} />
      </BsForm.Control> */}
      <Select
        label={i18n(field.label)}
        placeholder={i18n(field.placeholder)}
        onChange={handleInputChange(field)}
        items={items}
        disabled={disabled}
      />
      <FieldErrorMessage fieldErrors={fieldError} i18n={i18n} />
    </>
  );
};

const TextAreaInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
  i18n,
}: any) => {
  // const { disabled } = field;
  return (
    <>
      <Label>{i18n(field.label)}</Label>
      <div>Text area</div>
      <FieldErrorMessage fieldErrors={fieldError} i18n={i18n} />
    </>
  );
};

const TextInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
  i18n,
  questionnaire,
}: any) => {
  const { disabled } = field;
  const questionnaireField = () => {
    return fieldError ? (
      <div className="questionnaire__field-error">
        <FieldErrorMessage fieldErrors={fieldError} i18n={i18n} />
      </div>
    ) : (
      <div style={{ marginBottom: '4rem' }} />
    );
  };
  return (
    <>
      {questionnaire && field.title && (
        <div className="questionnaire__label">{i18n(field.title)}</div>
      )}
      {/* <Label>{field.label}</Label> */}
      {/* <BsForm.Control
        as={field.type}
        placeholder={field.placeholder}
        onChange={handleInputChange(field)}
        name={field.name}
        value={fieldValue}
        disabled={!!disabled}
        isInvalid={!!fieldError}
      /> */}
      <Input
        label={i18n(field.label)}
        aria-label="input-text"
        placeholder={i18n(field.placeholder)}
        onChange={handleInputChange(field)}
        value={fieldValue}
        disabled={!!disabled}
        validateStatus={fieldError ? 'error' : undefined}
      />
      {questionnaire ? (
        questionnaireField()
      ) : (
        <FieldErrorMessage fieldErrors={fieldError} i18n={i18n} />
      )}
    </>
  );
};

const DateInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
  i18n,
}: any) => {
  // const { disabled } = field;
  return (
    <>
      {/* {field.label && <Label>{field.label}</Label>} */}
      {/* {console.log('fieldValuesCheck', field, 'value', fieldValue)} */}
      {/* <input
        type="date"
        name={field.name}
        value={fieldValue}
        onChange={handleInputChange(field)}
        // onChhange={e => console.log("BS_DATE", e)}
        disabled={!!disabled}
      /> */}
      {field.label && <Label>{i18n(field.label)}</Label>}
      {/* <DatePicker
        name={field.name}
        value={fieldValue && moment(fieldValue)}
        onChange={handleInputChange(field)}
        disabled={!!disabled}
      /> */}
      <div>Date picker</div>
      <FieldErrorMessage fieldErrors={fieldError} i18n={i18n} />
    </>
  );
};

const FileInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
  i18n,
}: any) => {
  const { disabled } = field;
  return (
    <>
      {field.label && <Label>{i18n(field.label)}</Label>}
      <input
        type="file"
        name={field.name}
        value={fieldValue}
        onChange={handleInputChange(field)}
        disabled={!!disabled}
      />
      <FieldErrorMessage fieldErrors={fieldError} i18n={i18n} />
    </>
  );
};

const customForm = ({
  children,
  handleCancel,
  handleSubmit,
  handleBack,
  onSubmitLabel,
  title,
  twoColumns,
  props,
}: any) => {
  const customBack = {
    label: props.i18n('common.back'),
    onClick: () => handleBack(),
    // onClick: handleCancel ? cancel : undefined,
    uiType: 'secondary',
    alignIcon: 'start',
    withArrow: true,
  };

  return (
    <div className="p-4">
      <Form
        submitButton={{
          label: props.i18n(onSubmitLabel),
          withArrow: true,
          onClick: (e: React.FormEvent<HTMLFormElement>) => {
            handleSubmit(e, props);
          },
        }}
        backButton={handleBack && customBack}
      >
        <Form.Fieldset title={title} twoColumns={twoColumns}>
          {children}
        </Form.Fieldset>
      </Form>
      <button
        type="button"
        className={`questionnaire__cancel-button questionnaire__form-${
          handleBack ? `button` : `button-single`
        }`}
        onClick={() => handleCancel()}
      >
        {props.i18n('common.cancel')}
      </button>
    </div>
  );
};

const FieldGroup = ({ field, children, width }: any) => {
  return <div>{children}</div>;
};

const SectionForm = ({ title, twoColumns, children, i18n }: any) => {
  return (
    <Form.Fieldset title={i18n(title)} twoColumns={twoColumns}>
      {children}
    </Form.Fieldset>
  );
};

const customButton = ({ field, handleInputChange, i18n }: any) => {
  return (
    <>
      <Button
        aria-label="button-primary"
        size="small"
        label={i18n(field.label)}
        onClick={field.onClick}
      />
    </>
  );
};

export {
  Container,
  customForm as Form,
  FieldGroup,
  customButton as Button,
  CheckboxInput,
  RadioInput,
  TextAreaInput,
  TextInput,
  DateInput,
  SelectInput,
  SectionForm,
  FileInput,
  FieldErrorMessage,
};
