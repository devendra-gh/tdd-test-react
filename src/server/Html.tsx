import React, { Component } from 'react';
import serialize from 'serialize-javascript';

export interface HtmlProps {
  styles?: HtmlStyles[];
  scripts?: string[];
  scriptsHead?: string[];
  app: any;
  staticUrl: string;
  children?: string;
  mode?: string[];
  analyticsConfig: {
    key: string;
    host: string;
  };
}

export type HtmlStyles = {
  id: string;
  cssText: string;
};

/* global GIT_TAG */
const GIT = {
  TAG: GIT_TAG,
};

/**
 * Class responsible for page server side rendering (SSR)
 */
class Html extends Component<HtmlProps> {
  static defaultProps = {
    styles: [],
    scripts: [],
    scriptsHead: [],
    children: null,
    mode: [],
  };

  render() {
    const {
      app,
      staticUrl,
      children,
      styles,
      mode,
      analyticsConfig: { key, host },
    } = this.props;
    let { scripts, scriptsHead } = this.props;
    const stylesFromScripts = scripts!.filter(i => i.endsWith('.css'));
    scripts = scripts!.filter(i => !i.endsWith('.css'));
    scriptsHead = scriptsHead!.filter(i => !i.endsWith('.css'));
    const modeClassed = mode!.map(el => `mode__${el}`);

    return (
      <html className={[...modeClassed, 'no-js'].join(' ')} lang="en">
        <head>
          <meta name="release" content={GIT.TAG} />
          {scriptsHead.map(script => (
            <script key={script} src={`${script}?r=${GIT.TAG}`} />
          ))}
          {stylesFromScripts.map(link => (
            <link
              key={link}
              rel="stylesheet"
              type="text/css"
              href={`${link}?r=${GIT.TAG}`}
            />
          ))}
          <link
            rel="stylesheet"
            href={`${staticUrl}/ui-lib/colors-dynamic.css`}
          />
          {/* eslint-disable react/no-danger */}
          {styles &&
            styles.map(style => (
              <style
                key={style.id}
                id={style.id}
                dangerouslySetInnerHTML={{ __html: style.cssText }}
              />
            ))}
          {/* eslint-enable react/no-danger */}
        </head>
        <body>
          {/* eslint-disable react/no-danger */}
          {process.env.NODE_ENV === 'production' && (
            <script
              id="removeDevTool"
              type="text/javascript"
              src="scripts/removeDevTool.js"
            />
          )}
          <div id="app" dangerouslySetInnerHTML={{ __html: children || '' }} />
          <script
            id="staticData"
            type="json"
            dangerouslySetInnerHTML={{
              __html: serialize(app, { isJSON: true }),
            }}
          />
          {/* eslint-enable react/no-danger */}
          {scripts.map(script => (
            <script key={script} src={`${script}?r=${GIT.TAG}`} />
          ))}
          <script
            id="analyticsScript"
            src={`${staticUrl}/analytics/index.js?key=${key}&host=${host}`}
          />
        </body>
      </html>
    );
  }
}

export default Html;
