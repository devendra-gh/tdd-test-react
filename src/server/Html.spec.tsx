import React from 'react';
import { mount } from 'enzyme';

import Html from 'server/Html';

const requireProps = {
  title: 'some title',
  description: 'some description',
  app: {},
  url: 'some url',
  styles: [],
  scripts: [],
  staticUrl: '',
  analyticsConfig: {
    key: '',
    host: '',
  },
};

describe('Html', () => {
  it('renders', () => {
    expect(mount(<Html {...requireProps} />)).toMatchSnapshot();
  });

  it('renders with scripts and styles', () => {
    const scripts = [
      'a.css?r=GIT_TAG',
      'b.css?r=GIT_TAG',
      'c.js?r=GIT_TAG',
      'd.js?r=GIT_TAG',
    ];
    const styles = [
      {
        id: 'x',
        cssText: 'x content',
      },
      {
        id: 'y',
        cssText: 'y content',
      },
    ];
    const mode = ['mode1'];

    expect(
      mount(
        <Html
          {...requireProps}
          scripts={scripts}
          styles={styles}
          mode={mode}
        />,
      ),
    ).toMatchSnapshot();
  });
});
