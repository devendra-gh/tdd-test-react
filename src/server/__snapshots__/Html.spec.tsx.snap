// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Html renders 1`] = `
<Html
  analyticsConfig={
    Object {
      "host": "",
      "key": "",
    }
  }
  app={Object {}}
  description="some description"
  mode={Array []}
  scripts={Array []}
  scriptsHead={Array []}
  staticUrl=""
  styles={Array []}
  title="some title"
  url="some url"
>
  <html
    className="no-js"
    lang="en"
  >
    <head>
      <meta
        content="GIT_TAG"
        name="release"
      />
      <link
        href="/ui-lib/colors-dynamic.css"
        rel="stylesheet"
      />
    </head>
    <body>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
        id="app"
      />
      <script
        dangerouslySetInnerHTML={
          Object {
            "__html": "{}",
          }
        }
        id="staticData"
        type="json"
      />
      <script
        id="analyticsScript"
        src="/analytics/index.js?key=&host="
      />
    </body>
  </html>
</Html>
`;

exports[`Html renders with scripts and styles 1`] = `
<Html
  analyticsConfig={
    Object {
      "host": "",
      "key": "",
    }
  }
  app={Object {}}
  description="some description"
  mode={
    Array [
      "mode1",
    ]
  }
  scripts={
    Array [
      "a.css?r=GIT_TAG",
      "b.css?r=GIT_TAG",
      "c.js?r=GIT_TAG",
      "d.js?r=GIT_TAG",
    ]
  }
  scriptsHead={Array []}
  staticUrl=""
  styles={
    Array [
      Object {
        "cssText": "x content",
        "id": "x",
      },
      Object {
        "cssText": "y content",
        "id": "y",
      },
    ]
  }
  title="some title"
  url="some url"
>
  <html
    className="mode__mode1 no-js"
    lang="en"
  >
    <head>
      <meta
        content="GIT_TAG"
        name="release"
      />
      <link
        href="/ui-lib/colors-dynamic.css"
        rel="stylesheet"
      />
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "x content",
          }
        }
        id="x"
        key="x"
      />
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "y content",
          }
        }
        id="y"
        key="y"
      />
    </head>
    <body>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
        id="app"
      />
      <script
        dangerouslySetInnerHTML={
          Object {
            "__html": "{}",
          }
        }
        id="staticData"
        type="json"
      />
      <script
        key="a.css?r=GIT_TAG"
        src="a.css?r=GIT_TAG?r=GIT_TAG"
      />
      <script
        key="b.css?r=GIT_TAG"
        src="b.css?r=GIT_TAG?r=GIT_TAG"
      />
      <script
        key="c.js?r=GIT_TAG"
        src="c.js?r=GIT_TAG?r=GIT_TAG"
      />
      <script
        key="d.js?r=GIT_TAG"
        src="d.js?r=GIT_TAG?r=GIT_TAG"
      />
      <script
        id="analyticsScript"
        src="/analytics/index.js?key=&host="
      />
    </body>
  </html>
</Html>
`;
